src/lib/include/simd/simd_doubles.h

```diff
@@ -0,0 +1,162 @@
+#ifndef DISTOPIA_SIMD_DOUBLE_H
+#define DISTOPIA_SIMD_DOUBLE_H
+
+#include <cassert>
+#include <immintrin.h>
+
+// class to wrap __m128d 2 x packed doubles
+class SimdDoubleX2 {
+public:
+  inline void set(const double d);
+  inline void load(const double *source);
+  inline void store(double *target);
+  inline void loadU(const double *source);
+  inline void storeU(double *target);
+  inline void zero();
+  inline void reciprocal();
+
+  __m128d contents;
+};
+
+// set to all same value
+inline void SimdDoubleX2::set(const double d) { contents = _mm_set1_pd(d); }
+
+// aligned load from double*
+inline void SimdDoubleX2::load(const double *source) {
+  // check alignment
+  assert(size_t(source) % 16 == 0);
```

Comment:
[`uintptr_t`](https://en.cppreference.com/w/cpp/types/integer) is a more fitting type to cast to here.

Reply:
Use [`alignof`](https://en.cppreference.com/w/cpp/language/alignof) instead of hard-coding a constant.

Reply:
if we use loadU is redundant.